deployments/charts/kuma/templates/cp-service.yaml

```diff
@@ -14,28 +14,50 @@ metadata:
 spec:
   type: {{ .Values.controlPlane.service.type }}
   ports:
-    - port: 5680
+    - port: {{ .Values.controlPlane.diagnostics.service.port }}
       name: diagnostics
       appProtocol: http
-    - port: 5681
+      {{- if and (eq .Values.controlPlane.service.type "NodePort") .Values.controlPlane.diagnostics.service.nodePort }}
+      nodePort: {{ .Values.controlPlane.diagnostics.service.nodePort }}
+      {{- end }}
+    - port: {{ .Values.controlPlane.apiServer.http.service.port }}
       name: http-api-server
       appProtocol: http
-    - port: 5682
+      {{- if and (eq .Values.controlPlane.service.type "NodePort") .Values.controlPlane.apiServer.http.service.nodePort }}
+      nodePort: {{ .Values.controlPlane.apiServer.http.service.nodePort }}
+      {{- end }}
+    - port: {{ .Values.controlPlane.apiServer.https.service.port }}
       name: https-api-server
       appProtocol: http
+      {{- if and (eq .Values.controlPlane.service.type "NodePort") .Values.controlPlane.apiServer.https.service.nodePort }}
+      nodePort: {{ .Values.controlPlane.apiServer.https.service.nodePort }}
+      {{- end }}
     {{- if ne .Values.controlPlane.environment "universal" }}
-    - port: 443
+    - port: {{ .Values.controlPlane.admissionServer.service.port }}
```

Comment:
I think this would need to be reflected at https://github.com/kumahq/kuma/blob/69814a96abc636158a3bf6e4a15d8031db626e65/deployments/charts/kuma/templates/cp-webhooks-and-secrets.yaml#L66 with `port: ..`, if the port isn't 443.